editor/scene/gui/theme_editor_plugin.cpp

```diff
@@ -77,6 +77,24 @@ static void _rename_theme_type(EditorUndoRedoManager *p_ur, Theme *p_theme, cons
 	}
 }
 
+static String _validate_theme_type_name_input(const String &p_name) {
+	String type_name = p_name.strip_edges();
+
+	// An empty string is a special theme type.
+	if (type_name.is_empty()) {
+		return type_name; // Return early to avoid unnecessary COW.
+	}
```

Comment:
This part is a bit redundant, since in case of an empty name the loop will be empty too. I'm not saying that early checking is not needed, I'm just noting the inconsistency with `Theme::is_valid_type_name()`.

Reply:
This is to avoid triggering copy-on-write due to `ptrw()`. But indeed, copying an empty string shouldn't cause any harm :smile: